### PR TITLE
memoize all calls to selfExePath

### PR DESCRIPTION
**Pull request by nektro**
breaking out #9578 into multiple PRs in order to better pin down where the CI is going awry

### CODE REVIEW

File: src/bun.js/node/types.zig [hunk 1]
```diff
@@ -4935,7 +4934,7 @@ pub const Process = struct {
                 bun.String.static("bun"),
             );
         } else {
-            const exe_path = std.fs.selfExePathAlloc(allocator) catch null;
+            const exe_path = bun.selfExePath() catch null;
```
Comment (jdalton) [inline src/bun.js/node/types.zig:4937]: Above we handle errors with a pretty error for this. Here we `null`. Is the error likely an out of memory thing where we would do the `bun.outOfMemory()`?

Comment (nektro) [reply src/bun.js/node/types.zig:4937]: below this line there's an optional check and provide a fallback without needing to throw an error.

Comment (Jarred-Sumner) [reply src/bun.js/node/types.zig:4937]: its pretty unlikely for this to error, the current executable would have to have the permissions changed after it started

File: src/bun.zig [hunk 2]
```diff
@@ -2816,6 +2816,22 @@ pub fn linuxKernelVersion() Semver.Version {
...
+        var value: [4096]u8 = undefined;
+        var len: usize = 0;
+    };
+    if (memo.set) return memo.value[0..memo.len :0];
+    const init = try std.fs.selfExePath(&memo.value);
+    memo.len = init.len;
+    memo.value[memo.len] = 0;
+    memo.set = true;
+    return memo.value[0..memo.len :0];
+}
```
Comment (gvilums) [inline src/bun.zig]: Are we sure that `selfExePath` will only be called from one thread?

**Review by gvilums — COMMENTED** [review]
Looks good overall - see my concern about thread safety. If that's not an issue I think we should merge this.

File: src/bun.zig (hunk 2 above)
Comment (Jarred-Sumner) [reply src/bun.zig]: Fixed in c010351c3d46caba53e247a03a7c1f1840c3cee1

**Comment by Jarred-Sumner** [comment]
I believe the napi tests are new & unrelated.

**Pull request merged**